web/src/app/projects/edit/edit.component.ts

```diff
@@ -39,7 +44,14 @@ export class EditProjectComponent implements OnInit {
 
         this.projectSubscription = this.projectService
             .findOneById(this.uid)
-            .subscribe((project: ProjectModel) => this.projectForm.reset(project));
+            .subscribe((project: ProjectModel) => {
+                    this.project = project;
+                    this.projectForm.reset(project);
+                    // Added check for not authenticated user and private project details
+                    if ((project.type === 'private' && !this.authService.isAuthenticated) || !this.isAdmin()) {
+                        this.router.navigate(['/']);
```

Comment:
Is it possible to put this check in a `guard`?

Reply:
Or resolve?

Reply:
I have checked if we can create guard or not . 
We are checking project is public or private based upon the uid and then we are retrieving the project details from firebase but within guard we can not retrieve the through route without navigating to it. so we have to put a check in component itself for this.